x-pack/filebeat/module/suricata/eve/_meta/fields.yml

```diff
@@ -3,14 +3,752 @@
+
+      - name: state
+        type: keyword
+
+      - name: tcp_flags_ts
+        type: keyword
+
+      - name: rst
+        type: boolean
+
+      - name: fin
+        type: boolean
+
+    - name: fileinfo
+      type: group
+      fields:
+      - name: sha1
+        type: keyword
+
+      - name: filename
+        type: keyword
+
+      - name: tx_id
+        type: long
+
+      - name: state
+        type: keyword
+
+      - name: stored
+        type: boolean
+
+      - name: gaps
+        type: boolean
+
+      - name: sha256
+        type: keyword
+
+      - name: md5
+        type: keyword
+
+      - name: size
+        type: long
+
+    - name: icmp_type
+      type: long
+
+    - name: dest_port
+      type: long
+
+    - name: src_port
+      type: long
+
+    - name: proto
+      type: keyword
+
+    - name: pcap_cnt
+      type: long
+
     - name: src_ip
       type: ip
+
+    - name: dns
+      type: group
+      fields:
+      - name: type
+        type: keyword
+
+      - name: rrtype
+        type: keyword
+
+      - name: rrname
+        type: keyword
+
+      - name: rdata
+        type: keyword
+
+      - name: tx_id
+        type: long
+
+      - name: ttl
+        type: long
+
+      - name: rcode
+        type: keyword
+
+      - name: id
+        type: long
+
+    - name: flow_id
+      type: keyword
+
+    - name: email
+      type: group
+      fields:
+      - name: status
+        type: keyword
+
     - name: dest_ip
       type: ip
-    - name: flow_id
+
+    - name: icmp_code
+      type: long
+
+    - name: http
+      type: group
+      fields:
+      - name: status
+        type: long
+
+      - name: redirect
+        type: keyword
+
+      - name: http_user_agent
+        type: keyword
+
+      - name: protocol
+        type: keyword
+
+      - name: http_refer
+        type: keyword
+
+      - name: url
+        type: keyword
+
+      - name: hostname
+        type: keyword
+
+      - name: length
+        type: long
+
+      - name: http_method
+        type: keyword
+
+      - name: http_content_type
+        type: keyword
+
+    - name: timestamp
+      type: date
+
+    - name: in_iface
+      type: keyword
+
+    - name: alert
+      type: group
+      fields:
+      - name: category
+        type: keyword
+
+      - name: severity
+        type: long
+
+      - name: rev
+        type: long
+
+      - name: gid
+        type: long
+
+      - name: signature
+        type: keyword
+
+      - name: action
+        type: keyword
+
+      - name: signature_id
+        type: long
+
+    - name: ssh
+      type: group
+      fields:
+      - name: client
+        type: group
+        fields:
+        - name: proto_version
+          type: keyword
+
+        - name: software_version
+          type: keyword
+
+      - name: server
+        type: group
+        fields:
+        - name: proto_version
+          type: keyword
+
+        - name: software_version
+          type: keyword
+
+    - name: stats
```

Comment:
I'd be inclined to skip all of the integers under `stats`. They're already numerics in the JSON logs, and will therefore be autodetected as `long`.

Do we have a policy of trying to define as many fields as possible in modules (even when it's basically a no-op)? I understand that if the defaults change this could cause problems, but this seems like a lot of overhead for something that doesn't change the outcome.

Reply:
That's incorrect. This is just declaring these field types explicitly. They will all end up in ES the same whether this is there or not. Kibana will let you use them in dashboards etc regardless of this. If they are in an ElasticSearch index, you can use them in Kibana, there's no other dependency.

Reply:
As we just discussed, we'll keep stats and get rid of vars. Keeping stats around will ensure the Kibana index pattern is pre-populated with the numeric stats, which could eventually be used in visualizations we ship with the module. So it makes sense to pre-populate those with the index template.